NPE crash when received a strange intent from Firefox

- I have searched for existing issues that may be the same as or related to mine.
  - Possibly introduced by the commit: 410d759f36d582c23e8b5789e7462867750be801
  - Directly related issue then: #711
  - Indirectly related commit then: 1a6296eb513a0476fd27b81819611e05bc45c5d9

Please complete the following information:

    Device: 
    Android Version: 
    Orgzly Revived Version: 1.15.0
    Sync repository type (directory/Dropbox/WebDAV/Git): Dropbox

## Background

Firefox has this peculiar problem that some websites including DuckDuckGo fails to change the page title dynamically. It's left empty and defaults back to the page URL. Mozilla seems never fixing this issue. (Or some extension that I always use is causing this, I'm not sure. Never has such a thing happened on the desktop version, ~~I'm pretty sure~~ I guess.)

## What happens

When such a webpage is subject to sharing, the shared title is also left empty, while the page URL is shared just fine. When Orgzly Revived gets this, it closes in less than two seconds.

## What should happen

As with Orgzly, the recipient app never crashes. ~~The note title is simply left empty as it originally was, and only the URL is kept in the note title, or in Orgzly's case, in the note body.~~

## Data

The intent URI for such a share: `intent:#Intent;action=android.intent.action.SEND;type=text/plain;launchFlags=0x18080001;S.android.intent.extra.SUBJECT=;S.android.intent.extra.TEXT=https%3A%2F%2Fduckduckgo.com%2F%3Fq%3DFirefox%2BAndroid%2BDuckDuckGo%2Bpage%2Btitle%2Bempty%26t%3Dfpas%26ia%3Dweb;end`. Note the zero-length `S.android.intent.extra.SUBJECT`.

Stacktrace from logcat:
```
01-07 11:48:11.133  2423  2423 E AndroidRuntime: FATAL EXCEPTION: main
01-07 11:48:11.133  2423  2423 E AndroidRuntime: Process: com.orgzlyrevived, PID: 2423
01-07 11:48:11.133  2423  2423 E AndroidRuntime: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.orgzlyrevived/com.orgzly.android.ui.share.ShareActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.contains(java.lang.CharSequence)' on a null object reference
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2953)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3094)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:78)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1839)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:106)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:211)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:6775)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
01-07 11:48:11.133  2423  2423 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean java.lang.String.contains(java.lang.CharSequence)' on a null object reference
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at com.orgzly.android.ui.share.ShareActivity.M1(r8-map-id-8bc4c4a2799966b99d2f0d2733af98b6af9703bed9d0a1ef0b3841dc757fa8b5:57)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at com.orgzly.android.ui.share.ShareActivity.L1(r8-map-id-8bc4c4a2799966b99d2f0d2733af98b6af9703bed9d0a1ef0b3841dc757fa8b5:82)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at com.orgzly.android.ui.share.ShareActivity.onCreate(r8-map-id-8bc4c4a2799966b99d2f0d2733af98b6af9703bed9d0a1ef0b3841dc757fa8b5:19)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at android.app.Activity.performCreate(Activity.java:7169)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at android.app.Activity.performCreate(Activity.java:7159)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1271)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2933)
01-07 11:48:11.133  2423  2423 E AndroidRuntime:        ... 11 more
```

## Extra note with the workaround

https://github.com/orgzly-revived/orgzly-android-revived/blob/9d7612e1da24345b099e0139e97626221f0942e6/app/src/main/java/com/orgzly/android/ui/share/ShareActivity.java#L104-L113

Seems like this is the culprit? Introduced in 410d759f36d582c23e8b5789e7462867750be801, the workaround is to disable the "create org links from shared links" option, confirmed.

--- a/app/src/main/java/com/orgzly/android/ui/share/ShareActivity.java
+++ b/app/src/main/java/com/orgzly/android/ui/share/ShareActivity.java
@@ -105,17 +105,18 @@ public class ShareActivity extends CommonActivity
                 // Both "text" and "subject" received. Subject goes in heading, text goes in
                 // body.
                 String subject = intent.getStringExtra(Intent.EXTRA_SUBJECT);
+                data.content = intent.getStringExtra(Intent.EXTRA_TEXT);
                 if (subject != null && !subject.isEmpty()) {
                     data.title = subject;
-                }
-                data.content = intent.getStringExtra(Intent.EXTRA_TEXT);
-                if (AppPreferences.createOrgLinksFromSharedLinks(this) &&
-                        !data.title.contains("\n")) {
-                    try {
-                        new URI(data.content);
-                        data.title = "[[" + data.content + "][" + data.title + "]]";
-                        data.content = null;
-                    } catch (URISyntaxException ignored) {}
+                    if (AppPreferences.createOrgLinksFromSharedLinks(this) &&
+                            !data.title.contains("\n")) {
+                        // If content is a URL, turn the heading into a clickable link.
+                        try {
+                            new URI(data.content);
+                            data.title = "[[" + data.content + "][" + data.title + "]]";
+                            data.content = null;
+                        } catch (URISyntaxException ignored) {}
+                    }
                 }
             } else {
                 // A single text string was shared. Put it in heading or body, depending on the

